ts/src/profile-encoder.ts

```diff
@@ -0,0 +1,28 @@
+/**
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as pify from 'pify';
+import {gzip as gzipPromise} from 'zlib';
+
+import {perftools} from '../../proto/profile';
+
+const gzip = pify(gzipPromise);
+
+export async function encode(profile: perftools.profiles.IProfile):
+    Promise<Buffer> {
+  const buffer = perftools.profiles.Profile.encode(profile).finish();
```

Comment:
A sync version of similar code also exists in `index.ts`. It would be good to move that to this file now that we have a file specifically called `profile-encoder.ts`.

Reply:
https://github.com/google/pprof-nodejs/pull/14